Mathlib/Data/Finsupp/Defs.lean

```diff
@@ -463,4 +463,53 @@ theorem support_zipWith [D : DecidableEq α] {f : M → N → P} {hf : f 0 0 = 0
 
 end ZipWith
 
+section Restrict
+
+variable [Zero α] (f g : ι → α) (s : Finset ι)
```

Comment:
Is there a reason for `s` to be finite here?

Reply:
Ah, sorry, didn't notice you allowed any functions :facepalm:

Reply:
But then isn't this just a particular case of [Finsupp.indicator](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Data/Finsupp/Indicator.html#Finsupp.indicator) ?